taichi/analysis/same_statements.cpp

```diff
@@ -218,7 +245,7 @@ bool same_value(Stmt *stmt1, Stmt *stmt2) {
   // Note that we do not need to test !stmt2->common_statement_eliminable()
   // because if this condition does not hold,
   // same_statements(stmt1, stmt2) returns false anyway.
```

Comment:
Not a change in this PR but I liked the comment here since it alleviated my confusion :-)

Reply:
Actually this was correct when stmt1 and stmt2 share the same IR root, but now that we check their operands recursively, it may fail in this case:
```
$1 = rand()
$2 = const [1]
$3 = add $1 $2
...
$4 = rand()
$5 = const [1]
$6 = add $1 $2
```
$3 and $6 are the "same" even if they don't share the same IR root. They are also both `common_statement_eliminable()`. But they can have different values...

Reply:
Fixed in 3547548.